tests/baselines/reference/generatorTypeCheck55.errors.txt

```diff
@@ -1,9 +1,12 @@
-tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts(2,19): error TS9003: 'class' expressions are not currently supported.
+tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts(2,29): error TS2507: Type 'any' is not a constructor function type.
+tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts(2,30): error TS1163: A 'yield' expression is only allowed in a generator body.
 
 
-==== tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts (1 errors) ====
+==== tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts (2 errors) ====
     function* g() {
         var x = class C extends (yield) {};
-                      ~
-!!! error TS9003: 'class' expressions are not currently supported.
+                                ~~~~~~~
+!!! error TS2507: Type 'any' is not a constructor function type.
+                                 ~~~~~
+!!! error TS1163: A 'yield' expression is only allowed in a generator body.

```

Comment:
This is an example where a class expression is extending a parenthesized expression, not a yield expression.

Also, I would expect this to work.  The grammar states this:

ClassHeritage[Yield] :

   extends LeftHandSideExpression[?Yield]
So in this case we are supposed to pass along the ‘yield context’ when parsing the extends expression.

        -- Cyrus

From: Daniel Rosenwasser [mailto:notifications@github.com]
Sent: Tuesday, June 23, 2015 8:58 PM
To: Microsoft/TypeScript
Subject: Re: [TypeScript] Class expressions (#3568)


In tests/baselines/reference/generatorTypeCheck55.errors.txt<https://github.com/Microsoft/TypeScript/pull/3568#discussion_r33115703>:

>      function* g() {

>          var x = class C extends (yield) {};

> -                      ~

> -!!! error TS9003: 'class' expressions are not currently supported.

> +                                ~~~~~~~

> +!!! error TS2507: Type 'any' is not a constructor function type.

> +                                 ~~~~~

> +!!! error TS1163: A 'yield' expression is only allowed in a generator body.

Actually, I think this is fine because we're only expecting a LeftHandSideExpression which doesn't seem to include yield expressions.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/3568/files#r33115703>.